src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSigner.cs

```diff
@@ -5,23 +5,70 @@
 
 namespace System.Security.Cryptography.Cose
 {
+    /// <summary>
+    /// Provides signing information to be used with sign operations in <see cref="CoseSign1Message"/> and <see cref="CoseMultiSignMessage"/>.
+    /// </summary>
     public sealed class CoseSigner
     {
         internal readonly KeyType _keyType;
         internal readonly int? _algHeaderValueToSlip;
         internal CoseHeaderMap? _protectedHeaders;
         internal CoseHeaderMap? _unprotectedHeaders;
+
+        /// <summary>
+        /// Gets the private key to use during signing.
+        /// </summary>
+        /// <value>The private key to use during signing.</value>
         public AsymmetricAlgorithm Key { get; }
+
+        /// <summary>
+        /// Gets the hash algorithm to use to create the hash value for signing.
+        /// </summary>
+        /// <value>The hash algorithm to use to create the hash value for signing.</value>
         public HashAlgorithmName HashAlgorithm { get; }
+
+        /// <summary>
+        /// Gets the padding mode to use when signing.
+        /// </summary>
+        /// <value>The padding mode to use when signing.</value>
         public RSASignaturePadding? RSASignaturePadding { get; }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoseSigner"/> class.
+        /// </summary>
+        /// <param name="key">The private key to use for signing.</param>
+        /// <param name="hashAlgorithm">The hash algorithm to use to create the hash value for signing.</param>
+        /// <param name="protectedHeaders">The collection of protected header parameters to append to the message when signing.</param>
+        /// <param name="unprotectedHeaders">The collection of unprotected header parameters to append to the message when signing.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException">
+        ///   <para>
+        ///     <paramref name="key"/> is <see cref="RSA"/>, use <see cref="CoseSigner(RSA, RSASignaturePadding, HashAlgorithmName, CoseHeaderMap?, CoseHeaderMap?)"/> to specify a signature padding.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     <paramref name="key"/> is of an unsupported type.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     <see cref="CoseHeaderLabel.Algorithm"/> was incorrect based on the <paramref name="key"/> and <paramref name="hashAlgorithm"/>.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     <see cref="CoseHeaderLabel.Algorithm"/> was not contained in <paramref name="protectedHeaders"/>.
```

Comment:
Doesn't it add the header if it isn't present?

Reply:
Or does this actually mean

```suggestion
        ///     <paramref name="unprotectedHeaders"/> specifies a value with the <see cref="CoseHeaderLabel.Algorithm" /> label.
```

?

Reply:
this exception is thrown when the the header is incorrectly placed in the unproteted headers. Now that you point it out, it doesn't sound like it aligns with the reason I just described.

Reply:
> Or does this actually mean

yes